src/group.rs

```diff
@@ -1,134 +1,88 @@
-use async_stream::stream;
-use futures_core::stream::Stream;
 use serde::{Serialize, Deserialize};
-use eyre::{Result, anyhow};
+use eyre::{Result, Error, anyhow};
 use std::sync::Arc;
 use veilid_core::{
-    CryptoKey, DHTRecordDescriptor, CryptoTyped, VeilidUpdate, VeilidConfigInner, api_startup_config, CRYPTO_KIND_VLD0, Nonce, ProtectedStore, CryptoSystemVLD0, CryptoSystem, RoutingContext, KeyPair
+    CryptoKey, DHTRecordDescriptor, CryptoTyped, CryptoSystemVLD0, RoutingContext, SharedSecret, TypedKey
 };
 
-#[derive(Serialize, Deserialize)]
-pub struct GroupKeypair {
-    pub public_key: CryptoKey,
-    pub secret_key: Option<CryptoKey>,
-    pub encryption_key: CryptoKey,
-}
+use crate::common::DHTEntity;
+use crate::repo::Repo;
 
 #[derive(Clone)]
 pub struct Group {
     pub id: CryptoKey,
+    pub record_key: TypedKey,
```

Comment:
The DHTRecordDescriptor has a lot of this data already, I think we should do the following:

- remove `id/record_key/secret_key` from the Group struct
- have `id()` be a getter for `dht_Record.key`
- add a getter for `owner_key` and `owner_secret`
- Refactor the code to reference the group by the record key and pull the keypair out just for init
- the encryption key should stay the same

Reply:
Thank you for such clear guidelines! 

I’ve made the following updates:

- Removed id, record_key, and secret_key from the Group struct.
- Added id(), owner_key(), and owner_secret() as getters from dht_record.
- Refactored the code to reference the group by the record key and only pull the keypair for initialization.
- Kept the encryption key unchanged as requested.

That really helped cleaning it up.
Thanks again!